api/desecapi/models.py

```diff
@@ -195,209 +168,66 @@ def pdns_id(self):
 
         return name
 
-    # This method does not use @transaction.atomic as this could lead to
-    # orphaned zones on pdns.
-    def create_on_pdns(self):
-        """
-        Create zone on pdns
+    @property
+    def normalized_name(self):
+        return self.name.rstrip('.') + '.'
```

Comment:
We should guarantee that `Domain.name` never ends with a period. For that, we need to modify the regex expression in the field definition a little bit.

If we have that, we can just `return self.name + '.'`.